server/src/web/user/user.controller.ts

```diff
@@ -0,0 +1,46 @@
+import {
+    Body,
+    Controller,
+    Delete,
+    Get,
+    Inject,
+    Injectable,
+    Param, ParseIntPipe,
+    Post, Put,
+    Query,
+} from '@nestjs/common';
+import {User} from '../../domain/user/user.entity';
+import {UserDto} from '../../app/user/dto/user.dto';
+import {DeleteResult} from 'typeorm';
+import {UserService} from '../../app/user/user.service';
+
+@Injectable()
+@Controller('users')
+export class UserController {
+    constructor(@Inject('UserService') private service: UserService) {}
+
+    @Get(':id')
+    async get(@Param('id') id: number): Promise<User> {
+        return await this.service.get(id);
+    }
+    @Post()
+    async create(@Body() userDto: UserDto): Promise<User> {
+        return await this.service.create(userDto);
+    }
+    @Delete(':id')
+    async delete(@Param('id') id: number): Promise<DeleteResult> {
+        return await this.service.delete(id);
+    }
+    @Put(':id/increase-level')
```

Comment:
let's talk about url naming!

Reply:
i was not sure http method and url is good.

i referenced: [github-api](https://developer.github.com/v3/pulls/#update-a-pull-request-branch)
PUT /repos/:owner/:repo/pulls/:pull_number/update-branch

it was using verb in url

Reply:
How about it changes to `@Put(':id/level')` and `amount` has positive amount or negative amount?

Reply:
I dont understand positive amount or negative amount

Reply:
Then, how about `@Put(':id?level=amount')`?aaaaaa

Reply:
that's good but
maybe client will think like "set level with the amount".

like i wanted to say increase level 2 up but client will think like level = 2.

and there is another issue that one controller should route the url only with the parameter.
i mean increase

Reply:
about the comment `@Put(':id/level')`:


i think only resources should be used as 'noun'.
seeing [reference](https://restful-api-design.readthedocs.io/en/latest/resources.html),
it says resource is like object in oop. 

> A resource is an object with a type, associated data, relationships to other resources, and a set of methods that operate on it.

> resources has have data associated with them, 

so i think resource is like 'entity'.

but level is just value.